Revert "Revert D25199264: Enable callgrind collection for C++ snippets"

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48720 Revert "Revert D25199264: Enable callgrind collection for C++ snippets"**

This reverts commit 6646ff122d3215b77909f669fc26cf6a927030db.

Differential Revision: [D25273994](https://our.internmc.facebook.com/intern/diff/D25273994)

Commit:
Revert "Revert D25199264: Enable callgrind collection for C++ snippets"

This reverts commit 6646ff122d3215b77909f669fc26cf6a927030db.

[ghstack-poisoned]

mruberry commented:
Name your branch "ci-all/..." to run the failing test build in the github ci. Or push this commit to a new branch with that name.

Commit:
Update on "Revert "Revert D25199264: Enable callgrind collection for C++ snippets""


This reverts commit 6646ff122d3215b77909f669fc26cf6a927030db.

Differential Revision: [D25273994](https://our.internmc.facebook.com/intern/diff/D25273994)

[ghstack-poisoned]
